pkg/download/goget/goget.go

```diff
@@ -0,0 +1,179 @@
+package goget
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/download"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/storage"
+	"github.com/spf13/afero"
+)
+
+// New returns a download protocol by using
+// go get. You must have a modules supported
+// go binary for this to work.
+func New() download.Protocol {
+	return &goget{
+		goBinPath: env.GoBinPath(),
+		fs:        afero.NewOsFs(),
+	}
+}
+
+type goget struct {
+	goBinPath string
+	fs        afero.Fs
+}
+
+func (gg *goget) List(ctx context.Context, mod string) ([]string, error) {
+	const op errors.Op = "goget.List"
+	lr, err := gg.list(op, mod)
+	if err != nil {
+		return nil, err
+	}
+
+	return lr.Versions, nil
+}
+
+type listResp struct {
+	Path     string
+	Version  string
+	Versions []string `json:"omitempty"`
+	Time     time.Time
+}
+
+// dummyMod creates a random go module so that listing
+// a depdencny's upstream versions can run from the go cli.
+func dummyMod(fs afero.Fs, repoRoot string) error {
+	const op errors.Op = "goget.dummyMod"
+	gomodPath := filepath.Join(repoRoot, "go.mod")
+	gomodContent := []byte("module mod")
+	if err := afero.WriteFile(fs, gomodPath, gomodContent, 0666); err != nil {
+		return errors.E(op, err)
+	}
+	sourcePath := filepath.Join(repoRoot, "mod.go")
+	sourceContent := []byte(`package mod`)
+	if err := afero.WriteFile(fs, sourcePath, sourceContent, 0666); err != nil {
+		return errors.E(op, err)
+	}
+	return nil
+}
```

Comment:
this code would be super useful in the `Fetcher.Fetch` method. can we move it down to there and then make list use it?

Reply:
@arschles this is already there, I just copied it over. Should I export the function from that package?